### PR TITLE
Fix CSV breaking when new rows get added.

### PR DESCRIPTION
**Pull request by CakeVR**
## Summary
This PR fixes two bugs when updating CSV using the Translation feature.
- Adding a new translatable event breaks CSV file, if there are more than 2 locale keys.
- Updating the CSVs closed the currently opened timeline.

## Bug Cause

When a new line of text gets added to the CSV, eg. by adding a Text Event, it did not add trailing commas for each available locale.
Additionally, updating CSVs closed the current timeline node, but did not re-open it.

## Bug Fix

We count the number of locale entries and count each entry per CSV row.
For every missing entry, we add trailing commas.

We also save the resource at the beginning of the method and once finished, silently open it again.

## Additional Information

Changes comments and their positioning, adds a newlines were it helps readability.
In the future, this code section will require a full refactoring, as the overall method is quite complex.

### CODE REVIEW

File: addons/dialogic/Editor/Settings/settings_translation.gd [hunk 1]
```diff
@@ -231,24 +307,25 @@ func erase_translations() -> void:
...
+
 		for event in tml.events:
 			if event._translation_id:
 				event.remove_translation_id()
 				event.update_text_version()
+
 		tml.set_meta("timeline_not_saved", true)
-		ResourceSaver.save(tml, timeline_path)
-	
+		var result = ResourceSaver.save(tml, timeline_path)
```
Comment (Jowan-Spooner) [inline addons/dialogic/Editor/Settings/settings_translation.gd:321]: Why save the result (in an UNTYPED variable, naughty naughty) and then not use it?

File: addons/dialogic/Editor/Settings/settings_translation.gd [hunk 2]
```diff
@@ -66,136 +66,212 @@ func get_locales(filter:String) -> Dictionary:
...
-	settings_editor.editors_manager.clear_editor(settings_editor.editors_manager.editors['Timeline']['node'])
-	
-	# collect old lines in per project mode 
+	var csv_lines := []
+	# Contains already existing csv_lines as [key] = [value, value, ...] dict
+	var old_csv_lines := {}
+
+	var csv_columns := 0
+
+	var timeline_node = settings_editor.editors_manager.editors['Timeline']['node']
```
Comment (Jowan-Spooner) [inline addons/dialogic/Editor/Settings/settings_translation.gd:88]: Static typing please

File: addons/dialogic/Editor/Settings/settings_translation.gd [hunk 3]
```diff
@@ -66,136 +66,212 @@ func get_locales(filter:String) -> Dictionary:
...
+	# We will close this timeline to ensure it will properly update.
+	# By saving this reference, we can open it again.
+	var current_timeline = timeline_node.current_resource
```
Comment (Jowan-Spooner) [inline addons/dialogic/Editor/Settings/settings_translation.gd:91]: static typing

File: addons/dialogic/Editor/Settings/settings_translation.gd [hunk 4]
```diff
@@ -66,136 +66,212 @@ func get_locales(filter:String) -> Dictionary:
...
+
+				# In case there might be translations for this line already,
 				# add them at the end again (orig locale text is replaced).
 				if line[0] in old_csv_lines:
-					file.store_csv_line(line+Array(old_csv_lines[line[0]]).slice(2))
+					var old_line = old_csv_lines[line[0]]
+					var updated_line: PackedStringArray = line + Array(old_line).slice(2)
+
+					var line_columns: int = updated_line.size()
+					var line_columns_to_add = csv_columns - line_columns
```
Comment (Jowan-Spooner) [inline addons/dialogic/Editor/Settings/settings_translation.gd:174]: static typing please

File: addons/dialogic/Editor/Settings/settings_translation.gd [hunk 5]
```diff
@@ -66,136 +66,212 @@ func get_locales(filter:String) -> Dictionary:
...
+					# Add trailing commas to match the amount of columns.
+					for _i in range(line_columns_to_add):
+						updated_line.append("")
+
+					file.store_csv_line(updated_line)
 					counts[2] += 1
+
 				else:
+					var line_columns: int = line.size()
+					var line_columns_to_add = csv_columns - line_columns
```
Comment (Jowan-Spooner) [inline addons/dialogic/Editor/Settings/settings_translation.gd:185]: static typing please

File: addons/dialogic/Editor/Settings/settings_translation.gd [hunk 6]
```diff
@@ -66,136 +66,212 @@ func get_locales(filter:String) -> Dictionary:
...
-			# in case there might be translations for this line already, 
+			# in case there might be translations for this line already,
 			# add them at the end again (orig locale text is replaced).
 			if line[0] in old_csv_lines:
-				file.store_csv_line(PackedStringArray(line)+old_csv_lines[line[0]].slice(2))
+				var old_line: PackedStringArray = old_csv_lines[line[0]]
+				var updated_line: PackedStringArray = PackedStringArray(line)+old_line.slice(2)
+
+				var line_columns: int = updated_line.size()
+				var line_columns_to_add = csv_columns - line_columns
```
Comment (Jowan-Spooner) [inline addons/dialogic/Editor/Settings/settings_translation.gd:216]: static typing please

File: addons/dialogic/Editor/Settings/settings_translation.gd [hunk 7]
```diff
@@ -66,136 +66,212 @@ func get_locales(filter:String) -> Dictionary:
...
+				# Add trailing commas to match the amount of columns.
+				for _i in range(line_columns_to_add):
+					updated_line.append("")
+
+				file.store_csv_line(updated_line)
 				counts[2] += 1
+
 			else:
+				var line_columns: int = line.size()
+				var line_columns_to_add = csv_columns - line_columns
```
Comment (Jowan-Spooner) [inline addons/dialogic/Editor/Settings/settings_translation.gd:227]: static typing please

**Review by Jowan-Spooner — APPROVED** [review]

File: addons/dialogic/Editor/Settings/settings_translation.gd (hunk 1 above)
Comment (CakeVR) [reply addons/dialogic/Editor/Settings/settings_translation.gd:321]: Ooops, that is dead code I used to identify whether there is a saving issue when in Timeline text mode.
I did nothing but print it. Sadly, I didn't get an unused warning for this :/

**Pull request merged**